apps/files_trashbin/lib/Sabre/TrashbinPlugin.php

```diff
@@ -129,4 +132,40 @@ public function httpGet(RequestInterface $request, ResponseInterface $response):
 			$response->addHeader('Content-Disposition', 'attachment; filename="' . $node->getFilename() . '"');
 		}
 	}
+
+	/**
+	 * Check if a user has available space before attempting to
+	 * restore from trashbin unless they have unlimited quota.
+	 *
+	 * @param string $sourcePath
+	 * @param string $destinationPath
+	 * @return bool
+	 */
+	public function beforeMove(string $sourcePath, string $destinationPath): bool {
+		try {
+			$node = $this->server->tree->getNodeForPath($sourcePath);
+			$destinationNode = $this->server->tree->getNodeForPath(dirname($destinationPath));
```

Comment:
```suggestion
			$destinationNodeParent = $this->server->tree->getNodeForPath(dirname($destinationPath));
```
Just to make it clear that this is not the destination node itself but the parent folder.